### PR TITLE
JM dev branch 

### PR DESCRIPTION
**Pull request by JMuff22**
Some changes I've made to the repo

- Setup via anaconda and conda 
- Bugfixed the progress bar
- Updated Validate.sh help info
- Added the results for Silicon_SJ_ECP. 

You can merge any commits you find useful. I was unsure about adding the results for each of the examples so I just added the results for the ECP PP, others can be added in future or in the parent dir into the README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,18 +17,24 @@ Some extra things about the program
...
 ### Create local environment
 
-1) Clone the repository
-2) Go to repository
-3) 'pip install virtualenv'
-4) 'python -m venv virtualenv; source virtualenv/bin/activate'
-5) pip install -r requirements.txt
+1. Clone the repository
+2. Go to repository
+#### Pip
```
Comment (simulak1) [inline README.md:22]:
```suggestion
#### Setup environment with pip
```

File: README.md [hunk 2]
```diff
@@ -17,18 +17,24 @@ Some extra things about the program
...
+
+3. `pip install virtualenv`
+4. `python -m venv virtualenv; source virtualenv/bin/activate`
```
Comment (simulak1) [inline README.md:25]:
```suggestion
4. `python -m venv virtualenv; source virtualenv/bin/activate`
```
```suggestion
3.1. `python -m venv virtualenv; source virtualenv/bin/activate`
```

File: README.md [hunk 3]
```diff
@@ -17,18 +17,24 @@ Some extra things about the program
...
+5. pip install -r requirements.txt
+
+#### Anaconda
+
+3. `conda env create -f environment.yml`
```
Comment (simulak1) [inline README.md:30]:
```suggestion
#### Setup environment with pip

3.1. `pip install virtualenv`
3.2. `python -m venv virtualenv; source virtualenv/bin/activate`
3.3. `pip install -r requirements.txt`

#### Setup environment with Anaconda

3. `conda env create -f environment.yml`
```

File: examples/silicon_SJ_ECP/README [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+`./analyse.sh ../examples/silicon_SJ_ECP 192 2.1608567484189534E+03 2.5 5` 
+
+Polynomial degree: 5
+Mean squared error: 0.00640, cross-validation MSE: 0.07056
+--> BEST
+----------------------------------------
+PCF           : 0.9555
+STD           : 0.0011
+----------------------------------------
+Lifetime (ps) : 225.9338 <-core-corrected
```
Comment (simulak1) [inline examples/silicon_SJ_ECP/README:16]: On line 16, there is the text telling that the lifetime estimate has been core-corrected. On the level of the scripts and code, this means that as the script 'analyse.sh' in the 'scripts'-directory calls the python executable 'plyfit.py', it gives a value less than one (obtained from a two-component density functional theory calculation) to the argument 'corepart', as:

python ../positron_utils/pcf_fit/plyfit ... --corepart $COREPART_B15 ...

However, with the ECP pseudopotentials, the effect of the core is neglible. Thus, the corepart-argument should be set to 1 from within the 'analyse.sh'-script. Now the given value shows the actual lifetime estimate scaled down based on a corepart estimate from a calculation with AREP-pseudopotentals.

In conclusion, the existence of the README here is OK, but the values reported should be rewritten from analysis without core-corrections.

K.S.

**Review by simulak1 — CHANGES_REQUESTED** [review]
The commits are relevant and useful, only a couple of small changes required before merging, see suggestions and comments.

File: examples/silicon_SJ_ECP/README (hunk 4 above)
Comment (JMuff22) [reply examples/silicon_SJ_ECP/README]: Yes I forgot. The settings in `analyse.sh` will probably need to be changed then. Maybe a if statement containing the different fractional contributions from each source with an extra option for "no core correction" i.e `--corepart 1`. 

See: 

https://github.com/simulak1/polyfit_to_positron_qmc_data/pull/1/commits/3a775e4a549cf841736bc0013fb4e6934a7d602f

Comment (JMuff22) [reply examples/silicon_SJ_ECP/README]: Suggestion for the changes to `analyse.sh` for choosing a core correction option in https://github.com/simulak1/polyfit_to_positron_qmc_data/pull/1/commits/f7f8fc4222e3b5af0e219f9d9043cbd655e2cf1e. I think some explanation of the core contributions and citations should also be added to the `README`.

Future changes can also be made to lines 104-116 in `analyse.sh`. Currently they are not being used.

File: scripts/analyse.sh [hunk 5]
```diff
@@ -4,16 +4,18 @@
...
-    echo "./$0 <folder> <num-e> <volume> <fit_range> <max-pol> , "
+    echo "./$0 <folder> <num-e> <volume> <fit_range> <max-pol> <corepart> <element>, "
     echo "where"
     echo "<folder>:      found from local folder as system_size/opt_level,"
     echo "<num-e>: Number of electrons"
     echo "<volume>: Volume"
     echo "<fit_range>  : specified currently as fraction of the lattice constant"
     echo "<max-pol>    : maximum polynomial order"
+    echo "<corepart>    : core electron contribution modifier"
+    echo "<element>    : element selected"
```
Comment (simulak1) [inline scripts/analyse.sh:18]:
```suggestion
    echo "./$0 <folder> <num-e> <volume> <fit_range> <max-pol> <corepart>, "
    echo "where"
    echo "<folder>:      found from local folder as system_size/opt_level,"
    echo "<num-e>: Number of electrons"
    echo "<volume>: Volume"
    echo "<fit_range>  : specified currently as fraction of the lattice constant"
    echo "<max-pol>    : maximum polynomial order"
    echo "<corepart>    : core electron contribution modifier"
```

File: scripts/analyse.sh [hunk 6]
```diff
@@ -58,6 +60,47 @@ li_corepart_B15=0.000352885464408
...
+elif [ $6 = "B95" ] && [ $7 = "Li"  ]; then
+    corepart=$li_corepart_B95
+
+elif [ $6 = "KUR" ] && [ $7 = "Li"  ]; then
+    corepart=$li_corepart_KUR
+
+else 
+    echo "Core contribution option not found."
+    exit 5
+fi
```
Comment (simulak1) [inline scripts/analyse.sh:101]: It is a good idea to add flexibility to this script, and make the corepart into a cmd-line variable. However, I propose some changes before merging.

The corepart is unique to a given use case, and thus I think we should not include a list to this script. If one wants to add new data to the examples-dir, or use these scripts for some local data, the corepart would need to be set again. If we used your current change, the new corepart-variable would have to be written as a new option to the if-statement every time this script is applied to new data. 

Instead I suggest that we give the corepart as a cmd line option to the analyse.sh-script,  and create a new document to the scripts-folder where we list the coreparts for different materials with different functionals, and document the list appropriately. You can add this list before we merge, or I can do it after the merge. 


```suggestion
corepart=$6
```

File: scripts/analyse.sh (hunk 5 above)
Comment (simulak1) [reply scripts/analyse.sh:18]: Removed the <element>-option

File: scripts/analyse.sh [hunk 7]
```diff
@@ -4,16 +4,18 @@
 #SRC=<path-to-source-dir>
 SRC=../positron_utils/pcf_fit
 
-if [ "$#" -ne 5 ]; then
+if [ "$#" -ne 7 ]; then
```
Comment (simulak1) [inline scripts/analyse.sh:7]:
```suggestion
if [ "$#" -ne 6 ]; then
```

**Review by simulak1 — CHANGES_REQUESTED** [review]
With the proposed changes this can be merged. You can also add a separate list file for the coreparts in the scripts-folder, but this is not necessary at the moment.

File: examples/silicon_SJ_ECP/README (hunk 4 above)
Comment (simulak1) [reply examples/silicon_SJ_ECP/README]: I agree, but propose we don't use if-statement and give only the corepart as the cmd-line option to plyfit.py.

**Review by simulak1 — APPROVED** [review]
